explainer.md

```diff
@@ -349,84 +355,168 @@ input.accessibleNode.activeDescendant = optionList.accessibleNode;
+```
 
-Accessible Properties explicitly **will not** allow authors to access
-the _computed_ accessible node properties corresponding to a given DOM node:
+```js
+console.log(clickBtn.accessibleNode.role);   // "button"
+```
+
+#### "dirty value flag" model
+
+In order for the `accessibleNode` properties to always take precedence,
+while also reflecting ARIA attributes from the DOM,
+they will follow the
+[dirty value flag](https://html.spec.whatwg.org/multipage/forms.html#concept-fe-dirty)
+model used by `input`.
+
+This means that once a value has been written to an `accessibleNode`,
+the corresponding ARIA attribute *will never be read again*
+in the accessibility tree computation process,
+even if the value on `accessibleNode` is set to `null`.
 
 ```html
-<button id="composeBtn">Compose</button>
+<div id="promo" role="checkbox" aria-checked="true">Receive promotional offers</div>
 ```
+
 ```js
-var composeBtn = document.getElementById('composeBtn');
-console.log(composeBtn.accessibleNode.role);  // null, not "button"
-console.log(composeBtn.accessibleNode.label);  // null, not "Compose"
+console.log(promo.accessibleNode.role);  // "checkbox", reflected from DOM
+promo.accessibleNode.role = "switch";
+console.log(promo.accessibleNode.role);  // now "switch"
+promo.accessibleNode.role = null;
+console.log(promo.accessibleNode.role);  // now `null`
 ```
 
-This is analogous to the `style` attribute,
-which allows reading and setting an element's local style,
-while `getComputedStyle()` allows access to the full computed style.
+#### Reasoning
 
-[Phase 4](#phase-4-full-introspection-of-an-accessibility-tree) will allow accessing the computed properties.
+Again, more details reasoning is discussed in #60,
+but in summary the benefits are:
+- Reflecting from ARIA DOM attributes to `accessibleNode` means that
+authors can read ARIA attributes without using getAttribute(),
+which improves the ergonomics of ARIA;
+- The "dirty value flag" model results logically from the one-way reflection
+  from ARIA attributes to `accessibilityNode` properties,
+  while having `accessibilityNode` properties take strict precedence.
+  - Without this model,
+  it would be ambiguous what should happen when setting an `accessibilityNode` property
+  to `null`: should it stay `null`, or go back to reflecting the ARIA attribute?
+  - And if the latter, how would you ever override the ARIA attribute to get back to a default value?
 
 #### Validation
 
-Querying an Accessible Property can be used as a limited form of validation and feature detection.
+The ARIA spec defines a
+[finite list of value types](https://www.w3.org/TR/wai-aria-1.1/#propcharacteristic_value)
+which a particular ARIA attribute may take.
+
+Where these types have a clear 
+[mapping](https://www.w3.org/TR/wai-aria-1.1/#typemapping) 
+on to a basic JavaScript type,
+for example "true/false" which maps directly to a boolean type,
+the `accessibleNode` property can be set to that type.
```

Comment:
The way this paragraph is written, I think someone could interpret it
as saying that you *can* use a primitive JavaScript type, but you
don't have to, even though the example makes it more clear below.

How about:

Unlike in ARIA, where all attributes are strings, AccessibleNode properties
have more specific types that are more expressive and more clearly
express their semantics. For example, since aria-disabled only takes
the values "true" and "false", AccessibleNode.disabled is of type
boolean, and since aria-rowindex takes an integer, AccessibleNode.rowIndex
is of type long.